src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java

```diff
@@ -0,0 +1,415 @@
+                                             CitationFetcher.SearchType searchType, Button importButton,
+                                             ProgressIndicator progress, List<BibEntry> fetchedList,
+                                             ObservableList<CitationRelationItem> observableList) {
+        hideNodes(abortButton, progress);
+
+        observableList.setAll(fetchedList.stream().map(entr -> new CitationRelationItem(entr, false))
+                .collect(Collectors.toList()));
+
+        if (!observableList.isEmpty()) {
+            listView.refresh();
+        } else {
+            Label placeholder = new Label(Localization.lang("No articles found"));
+            listView.setPlaceholder(placeholder);
+        }
+        BooleanBinding booleanBind = Bindings.isEmpty(listView.getCheckModel().getCheckedItems());
+        importButton.disableProperty().bind(booleanBind);
+        importButton.setOnMouseClicked(event -> importEntries(listView.getCheckModel().getCheckedItems(), searchType, entry));
+        showNodes(refreshButton, importButton);
+    }
+
+    private void prepareToSearchForRelations(Button abortButton, Button refreshButton, Button importButton,
+                                             ProgressIndicator progress, BackgroundTask<List<BibEntry>> task) {
+        showNodes(abortButton, progress);
+        hideNodes(refreshButton, importButton);
+
+        abortButton.setOnAction(event -> {
+            hideNodes(abortButton, progress, importButton);
+            showNodes(refreshButton);
+            task.cancel();
+            dialogService.notify(Localization.lang("Search aborted!"));
+        });
+    }
+
+    private void hideNodes(Node... nodes) {
+        Arrays.stream(nodes).forEach(node -> node.setVisible(false));
+    }
+
+    private void showNodes(Node... nodes) {
+        Arrays.stream(nodes).forEach(node -> node.setVisible(true));
+    }
+
+    // Absolute-phase phenomena in photoionization with few-cycle laser pulses
+
+    /**
+     * Function to import selected entries to the database. Also writes the entries to import to the CITING/CITED field
+     *
+     * @param entriesToImport entries to import
+     */
+    private void importEntries(List<CitationRelationItem> entriesToImport, CitationFetcher.SearchType searchType, BibEntry entry) {
+        citingTask.cancel();
+        citedByTask.cancel();
+        List<BibEntry> entries = entriesToImport.stream().map(CitationRelationItem::getEntry).collect(Collectors.toList());
+        ImportHandler importHandler = new ImportHandler(
+                databaseContext,
+                preferencesService,
+                fileUpdateMonitor,
+                undoManager,
+                stateManager,
+                dialogService,
+                Globals.TASK_EXECUTOR);
+        importHandler.importEntries(entries);
+
+        dialogService.notify(Localization.lang("Number of entries successfully imported") + ": " + entriesToImport.size());
+    }
```

Comment:
I have the strong suspicion, that some of this code belongs into a ViewModel

Reply:
It surely does, but I thought we release the feature now letting users try it out and do some refactoring while working on the known limitations in later PRs.

Reply:
ok, im fine with that. let ua just not forget about this, maybe an issue in the maintainers focus or sthg